native/packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js

```diff
@@ -44,37 +49,117 @@ const styles = StyleSheet.create({
   },
 });
 
-const BpkHorizontalNav = (props) => {
-  const {
-    children,
-    spaceAround,
-    style,
-    ...rest
-  } = props;
-  const navStyle = [styles.horizontalNav];
-  if (spaceAround) {
-    navStyle.push(styles.spaceAround);
+const validateSelectedIndex = (selectedIndex, childrenCount) => {
+  if (typeof selectedIndex !== 'number' || !Number.isInteger(selectedIndex)) {
+    return false;
+  }
+  if (selectedIndex < 0 || selectedIndex >= childrenCount) {
+    return false;
   }
-  return (
-    <ScrollView
-      alwaysBounceHorizontal={false}
-      contentContainerStyle={[navStyle, style]}
-      horizontal
-      showsHorizontalScrollIndicator={false}
-      {...rest}
-    >
-      { children }
-    </ScrollView>
-  );
+  return true;
 };
 
+class BpkHorizontalNav extends React.Component {
+  constructor(props) {
+    super(props);
+    if (props.selected && !validateSelectedIndex(props.selected, React.Children.count(props.children))) {
```

Comment:
this shouldn't be in ctor, it wil only fire once and won't capture dynamic updates to children